chain/filecoin/filecoin_test.go

```diff
@@ -0,0 +1,126 @@
+package filecoin_test
+
+import (
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	filaddress "github.com/filecoin-project/go-address"
+	filtypes "github.com/filecoin-project/lotus/chain/types"
+	"github.com/ipfs/go-cid"
+	"github.com/renproject/id"
+	"github.com/renproject/multichain"
+	"github.com/renproject/multichain/chain/filecoin"
+	"github.com/renproject/pack"
+	"github.com/renproject/surge"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Filecoin", func() {
+	Context("when broadcasting a tx", func() {
+		It("should work", func() {
+			// create context for the test
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			// instantiate the client
+			client, err := filecoin.NewClient(
+				filecoin.DefaultClientOptions().
+					WithAddress("ws://127.0.0.1:1234/rpc/v0").
+					WithAuthToken("Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJBbGxvdyI6WyJyZWFkIiwid3JpdGUiLCJzaWduIiwiYWRtaW4iXX0.673MLa4AmbhNeC1Hj2Bn6c4t_ci68I0amkqAEHea8ik"),
+			)
+			Expect(err).ToNot(HaveOccurred())
+
+			// read the private key that we will send transactions from
+			senderPrivKeyStr := os.Getenv("FILECOIN_PK")
+			if senderPrivKeyStr == "" {
+				panic("FILECOIN_PK is undefined")
+			}
+			var ki filtypes.KeyInfo
+			data, err := hex.DecodeString(senderPrivKeyStr)
+			Expect(err).NotTo(HaveOccurred())
+			err = json.Unmarshal(data, &ki)
+			Expect(err).NotTo(HaveOccurred())
+			senderPrivKey := id.PrivKey{}
+			err = surge.FromBinary(&senderPrivKey, ki.PrivateKey)
+			Expect(err).NotTo(HaveOccurred())
+
+			// read sender's address into the filecoin-compatible format
+			senderAddr := os.Getenv("FILECOIN_ADDRESS")
+			if senderAddr == "" {
+				panic("FILECOIN_ADDRESS is undefined")
+			}
+			senderFilAddr, err := filaddress.NewFromString(string(senderAddr))
+			Expect(err).NotTo(HaveOccurred())
+
+			// FIXME
+			// random recipient
+			recipientPK := id.NewPrivKey()
+			recipientPubKey := recipientPK.PubKey()
+			recipientPubKeyCompressed, err := surge.ToBinary(recipientPubKey)
+			Expect(err).NotTo(HaveOccurred())
+			recipientFilAddr, err := filaddress.NewSecp256k1Address(recipientPubKeyCompressed)
+			Expect(err).NotTo(HaveOccurred())
+
+			// construct the transaction builder
+			gasFeeCap := pack.NewU256FromU64(pack.NewU64(149838))
+			gasLimit := pack.NewU256FromU64(pack.NewU64(495335))
+			gasPremium := pack.NewU256FromU64(pack.NewU64(149514))
+			filTxBuilder := filecoin.NewTxBuilder(gasFeeCap, gasLimit, gasPremium)
+
+			// build the transaction
+			amount := pack.NewU256FromU64(pack.NewU64(100000000))
+			nonce := pack.NewU256FromU64(pack.NewU64(0))
+			payload := pack.Bytes(nil)
+			tx, err := filTxBuilder.BuildTx(
+				multichain.Address(pack.String(senderFilAddr.String())),
+				multichain.Address(pack.String(recipientFilAddr.String())),
+				amount, nonce,
+				pack.U256{}, pack.U256{},
```

Comment:
As per the other comment, we should be passing in gas limit/price and using it to infer limit/cap.